src/functions/definitions/base_function_test.ts

```diff
@@ -0,0 +1,97 @@
+import { assertEquals } from "../../dev_deps.ts";
+import Schema from "../../schema/mod.ts";
+import { FunctionDefinition } from "./base_function.ts";
+
+import {
+  ManifestFunctionSchema,
+  ManifestFunctionType,
+} from "../../manifest/manifest_schema.ts";
+import {
+  ParameterSetDefinition,
+  PossibleParameterKeys,
+} from "../../parameters/types.ts";
+import { FunctionDefinitionArgs } from "../types.ts";
+
+export class TestBaseFunctionDefinition<
```

Comment:
nit: I wonder how valuable it is to make `FunctionDefinition` an abstract class when we need to do something like this to test.

Theoretically the majority of our tests should be testing the base `FunctionDefinition` functionality, and tests on the extended classes should focus on the extensions since we validated what `FunctionDefinition` does

Reply:
I've removed the use of a `base` abstract class
- The logic was moved to SlackFunctionDefinition
- ConnectorDefinitions are not included in the Manifest and don't require this logic